### PR TITLE
Complete AcadeMix DAO layer and APIs to manage languages

### PR DESCRIPTION
**Pull request by YohanAvishke**
## Purpose
This PR is a part of https://github.com/sef-global/sef-core/issues/35

## Goals
New APIs, Service level methods to manage languages, and DAO layer to manage all AcadeMix data.

## Approach
- Updated MultiverseApis to manage Languages
- Added new Entities to manage all AcadeMix data
- Added a new service and repository to manage languages

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: backend/src/main/java/org/sefglobal/core/service/LanguageService.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     * @param localeIdentifier which is the LCID
+     * @throws BadRequestException if validating the {@code localeIdentifier} fails
+     */
+    public void addLanguage(String localeIdentifier) throws BadRequestException {
+        if (localeIdentifier == null || localeIdentifier.isEmpty()) {
+            String msg = "Error, LCID: " + localeIdentifier + " is invalid.";
+            log.error(msg);
+            throw new BadRequestException(msg);
+        }
+
```
Comment (jayasanka-sack) [inline backend/src/main/java/org/sefglobal/core/service/LanguageService.java:50]: Remove whitespaces

File: backend/src/main/java/org/sefglobal/core/model/SubCategoryTranslation.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    @OneToOne(cascade = CascadeType.ALL)
+    @JoinColumn(
+            name = "language_id",
+            referencedColumnName = "id",
+            nullable = false
+    ) private Language language;
+
+    private String name;
+
```
Comment (jayasanka-sack) [inline backend/src/main/java/org/sefglobal/core/model/SubCategoryTranslation.java:40]: Remove whitespaces

Comment (YohanAvishke) [reply backend/src/main/java/org/sefglobal/core/model/SubCategoryTranslation.java:40]: et's keep this here space before the end of class

File: backend/src/main/java/org/sefglobal/core/service/LanguageService.java (hunk 1 above)
Comment (YohanAvishke) [reply backend/src/main/java/org/sefglobal/core/service/LanguageService.java]: Fixed in https://github.com/sef-global/sef-core/commit/47efce4620aefe85a883da2cdd8defe96a0f5dd0

File: backend/src/main/java/org/sefglobal/core/model/SubCategoryTranslation.java (hunk 2 above)
Comment (jayasanka-sack) [reply backend/src/main/java/org/sefglobal/core/model/SubCategoryTranslation.java:40]: okay!

**Pull request merged**